pkg/loop/internal/median.go

```diff
@@ -186,8 +194,20 @@ func newMedianProviderClient(b *brokerExt, cc grpc.ClientConnInterface) *medianP
 	m.reportCodec = &reportCodecClient{b, pb.NewReportCodecClient(m.cc)}
 	m.medianContract = &medianContractClient{pb.NewMedianContractClient(m.cc)}
 	m.onchainConfigCodec = &onchainConfigCodecClient{b, pb.NewOnchainConfigCodecClient(m.cc)}
-	m.chainReader = &chainReaderClient{b, pb.NewChainReaderClient(m.cc)}
-	m.codec = &codecClient{b, pb.NewCodecClient(m.cc)}
+	anyRetVal := 10
+
+	maybeCr := &chainReaderClient{b, pb.NewChainReaderClient(m.cc)}
+	err := maybeCr.GetLatestValue(context.Background(), types.BoundContract{}, "", nil, &anyRetVal)
+	if status.Convert(err).Code() != codes.Unknown {
+		m.chainReader = maybeCr
+	}
+
+	maybeCodec := &codecClient{b, pb.NewCodecClient(m.cc)}
+	err = maybeCodec.Decode(context.Background(), []byte{}, &anyRetVal, "")
+	if status.Convert(err).Code() != codes.Unknown {
+		m.codec = maybeCodec
+	}
+
```

Comment:
Is calling a 'real' service the only option here? Or is there some way to interact with the `pb.NewCodecClient` directly instead?

Reply:
I couldn't find any better way, it's not supported out of the box for grpc.  I asked bing chat and it suggested doing this too :(.